x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Realm.java

```diff
@@ -56,6 +57,21 @@ public int order() {
         return config.order;
     }
 
+    /**
+     * Each Realm can define its own auth scheme challenge to be sent in
+     * 'WWW-Authenticate' header
+     * <p>
+     * By default the auth scheme is 'Basic', realms like Kerberos can use
+     * 'Negotiate' scheme or OAuth could use 'Bearer' scheme or custom realms with
+     * custom auth scheme as defined in
+     * <a href="https://tools.ietf.org/html/rfc7235#section-4.1">RFC7235</a><br>
+     *
+     * @return value to be returned in 'WWW-Authenticate' response header.
+     */
+    public String getWWWAuthenticateHeaderValue() {
```

Comment:
I know I normally lean towards "just build what you need", but this method feels way too specific to me.
Did you consider a `getAuthenticationFailureHeaders()` method instead?

Reply:
> Did you consider a getAuthenticationFailureHeaders() method instead?

+1. I feel like this is the way to go. Ultimately this might allow us to remove the authentication failure handler (way off low priority future idea)